include/rocksdb/file_system.h

```diff
@@ -722,7 +740,9 @@ struct FSReadRequest {
   size_t len;
 
   // A buffer that MultiRead()  can optionally place data in. It can
-  // ignore this and allocate its own buffer
+  // ignore this and allocate its own buffer.
+  // The lifecycle of scratch should be until IO is completed. In case of
+  // asynchronous reads, it should be maintained until callback has been called.
```

Comment:
I would clearly state that's scratch is allocated by RocksDB. Also, change "should" -> "will", to clarify that the FS doesn't have to worry about it.